src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -394,3 +412,1107 @@ def normalize(self, archive, logger):
+            conversions_results = []
+            for i in self.results[0].reactants_conversions:
+                if i.name in ['He', 'helium', 'Ar', 'argon', 'inert']:
+                    continue
+                else:
+                    for j in self.reaction_conditions.reagents:
+                        if i.name == j.name:
+                            if j.pure_component.iupac_name is not None:
+                                i.name = j.pure_component.iupac_name
+                            if i.gas_concentration_in is None:
+                                i.gas_concentration_in = j.gas_concentration_in
+                            react = Reactant(
+                                name=i.name,
+                                conversion=i.conversion,
+                                gas_concentration_in=i.gas_concentration_in,
+                                gas_concentration_out=i.gas_concentration_out,
+                            )
+                            conversions_results.append(react)
+                            if (
+                                np.allclose(
+                                    i.gas_concentration_in, j.gas_concentration_in
+                                )
+                                is False
+                            ):
+                                logger.warn(f"""Gas concentration of '{i.name}' is not
+                                            the same in reaction_conditions and
+                                            results.reactants_conversions.""")
+        return conversions_results
+
+    def check_sample(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        if self.samples:
+            if self.samples[0].lab_id is not None and self.samples[0].reference is None:
+                sample = CompositeSystemReference(
+                    lab_id=self.samples[0].lab_id, name=self.samples[0].name
+                )
+                sample.normalize(archive, logger)
+                self.samples = []
+                self.samples.append(sample)
+            if self.samples[0].reference is not None:
+                self.populate_catalyst_sample_info(archive, logger)
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+        self.normalize_reaction_conditions(archive, logger)
+
+        self.populate_reactivity_info(archive, logger)
+        self.check_sample(archive, logger)
+
+        if self.results is None or self.results == []:
+            return
+
+        self.results[0].normalize(archive, logger)
+        if len(self.results) > 1:
+            logger.warn(
+                """Several instances of results found. Only the first result
+                is considered for normalization."""
+            )
+        conversions_results = self.return_conversion_results(archive, logger)
+
+        product_results = []
+        if self.results[0].products is not None:
```

Comment:
I *think* repeating nomad sections are usually empty lists that can be iterated over directly but I might be wrong.